packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java

```diff
@@ -310,4 +308,87 @@ public void linkToParentNodeWithRules()
                 .get(0).onModel().assertThat().isEqualTo(expectedRuleSet);
     }
 
+    /**
+     * Check we can DELETE/unlink a ruleset
+     *
+     * DELETE /nodes/{folderNodeId}/rule-set-links/{ruleSetId}.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES})
+    public void unlinkRuleSet()
+    {
+        STEP("Create folders in existing site");
+        final FolderModel ruleFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+        final FolderModel folder = dataContent.usingUser(user).usingSite(site).createFolder();
+
+        STEP("Create a rule in the rule folder.");
+        RestRuleModel ruleModel = createRuleModel("ruleName");
+        RestRuleModel rule = restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet()
+                .createSingleRule(ruleModel);
+
+        STEP("Get the rule sets for the folder and find the rule set id");
+        final RestRuleSetModelsCollection ruleSets = restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder)
+                .getListOfRuleSets();
+        ruleSets.assertThat().entriesListCountIs(1);
+        final String ruleSetId = ruleSets.getEntries().get(0).onModel().getId();
+
+        STEP("Link to a rule folder");
+        final RestRuleSetLinkModel request = new RestRuleSetLinkModel();
+        request.setId(ruleFolder.getNodeRef());
+        final RestRuleSetLinkModel ruleLink = restClient.authenticateUser(user).withCoreAPI().usingNode(folder).createRuleLink(request);
+
+        STEP("Unlink the rule set");
```

Comment:
Before unlinking you didn't check if the link really exists (linking result code or if linkedTo is true).

Reply:
Fair enough - I agree